lib/text/ui_text_displayer.js

```diff
@@ -65,12 +65,16 @@ shaka.text.UITextDisplayer = class {
 
     this.videoContainer_.appendChild(this.textContainer_);
 
+    /** @private {number} */
+    const updatePeriod = (config && config.captionsUpdatePeriod) ?
```

Comment:
We have to detect that we do not have a configuration and show a deprecate message:

```
if (!config || !config.captionsUpdatePeriod) {
  shaka.Deprecate.deprecateFeature(5,
      'UITextDisplayer w/ config',
      'Please migrate from initializing UITextDisplayer with a config.');
}
```

Reply:
I'm going to try to add this to the PR so I can merge it and release 4.8.2 asap.